src/app/chat/chat-room/chat-room.component.ts

```diff
@@ -95,58 +162,27 @@ export class ChatRoomComponent implements OnInit {
       // remove typed message from text field.
       this.message = ''; 
       // createing prams need to send message
-      let data = {
-        message: message,
-        team_id: this.selectedChat.team_id,
-        to: null,
-      };
+      let data:any;
       if (this.selectedChat.is_team) {
-        data.to = "team";
-      } else {
-        data.to = this.selectedChat.team_member_id;
-      }
-      this.chatService.postNewMessage(data).subscribe((response) => {
-        this.messageList.push(response);
-      }, (error) => {});
-    }
-  }
-
-  private updateMessageListResponse(response, loadMore): void {
-    let index = 0;
-    let tempRes = null;
-    if (response.length > 0) {
-      this.messageList = [];
-      for (index = 0; index < response.length; index++) {
-        if (response[index] && !response[index].is_sender) {
-          if (this.selectedChat.is_team) {
-            this.getValidChatColors(this.chatColors, response, index);
-          } else {
-            response[index].chat_color = this.selectedChat.chat_color;
-          }
+        data = {
+          message: message,
+          team_id: this.selectedChat.team_id,
+          to: "team",
+          participants_only: this.selectedChat.participants_only
         }
-        if (index === response.length - 1) {
-          tempRes = Object.assign([], response);
-          tempRes.reverse();
-          if (loadMore) {
-            this.messageList = tempRes.concat(this.messageList);
-          } else {
-            this.messageList = tempRes;
-            this.content.scrollToBottom();
-          }
-          this.markAsSeen(tempRes);
+      } else {
+        data = {
+          message: message,
+          team_id: this.selectedChat.team_id,
+          to: this.selectedChat.team_member_id
         }
       }
-    }
-  }
-
-  private getValidChatColors(chatColors, response, index) {
-    let chatcolor = chatColors.find(function(chat) {
-      return chat.name === response[index].sender_name;
-    });
-    if (chatcolor) {
-      response[index].chat_color = chatcolor.chat_color;
-    } else {
-      response[index].chat_color = this.chatService.getRandomColor();
+      this.chatService.postNewMessage(data).subscribe(
+        response => {
+          this.messageList.push(response.data);
```

Comment:
User should be able to see the message he/she is sending directly without delay.
We can display a spinner besides the message indicating that it is sending to the server.